src/dmd/mars.d

```diff
@@ -815,9 +813,8 @@ extern (C++) void generateJson(Modules* modules)
             jsonfilename = FileName.forceExt(n, global.json_ext);
         }
         ensurePathToNameExists(Loc.initial, jsonfilename);
-        auto jsonfile = new File(jsonfilename);
-        jsonfile.setbuffer(buf.data, buf.offset);
-        jsonfile._ref = 1;
+        auto jsonfile = File(jsonfilename);
+        jsonfile.hijackBufferFrom(buf);
```

Comment:
Ditto; `buf` is destructed right after `jsonFile`, so just transfer ownership (and keep the `File` on the stack).